recipe/meta.yaml

```diff
@@ -38,19 +33,15 @@ requirements:
     - {{ cdt('libxau') }}               # [linux]
     - {{ cdt('libxi') }}                # [linux]
     - {{ cdt('libxxf86vm') }}           # [linux]
-    - {{ cdt('libxshmfence-devel') }}    # [linux and ppc64le]
     - cmake
+    - ninja                             # [win]
     - make                              # [not win]
-    - patch                             # [not win]
-    - m2-patch                          # [win]
   host:
-    - jpeg
-    - libglu                            # [linux]
-    - freeglut                          # [linux or win]
-    - pcre  # [ppc64le]
+    - jpeg {{ jpeg }}
+    - libglu 9.0.0                     # [linux]
+    - freeglut 3.0.0                    # [linux or win]
   run:
-    - jpeg
-    - pcre  # [ppc64le]
+    - jpeg # pin through run_exports
     - {{ pin_compatible('libglu') }}    # [linux]
     - {{ pin_compatible('freeglut') }}  # [linux or win]
```

Comment:
I'm not too sure why, but on Windows conda-build complains that `freeglut` is not used.
I see
```
2024-04-22T14:16:56.53742+00:00   | INFO     |   -- Found GLUT: C:/b/abs_855qth_2vi/croot/jasper_1713795313221/_h_env/Library/lib/freeglut.lib
2024-04-22T14:16:56.563617+00:00  | INFO     |   -- Looking for include file GL/glut.h
2024-04-22T14:16:56.580031+00:00  | INFO     |   -- Looking for include file GL/glut.h - not found
2024-04-22T14:16:56.596432+00:00  | INFO     |   -- Looking for include file GLUT/glut.h
2024-04-22T14:16:56.612045+00:00  | INFO     |   -- Looking for include file GLUT/glut.h - not found
2024-04-22T14:16:56.62765+00:00   | INFO     |   -- Looking for include file glut.h
2024-04-22T14:16:56.654385+00:00  | INFO     |   -- Looking for include file glut.h - not found
```
which is suspicious.
I also see
```
2024-04-22T14:16:35.792548+00:00  | INFO     |   CMAKE_REQUIRE_INCLUDES C:/b/abs_855qth_2vi/croot/jasper_1713795313221/_h_env/Library/include
2024-04-22T14:16:36.431289+00:00  | INFO     |   CMake Warning at CMakeLists.txt:716 (message):
2024-04-22T14:16:36.458164+00:00  | INFO     |     The glut.h header file is missing.
2024-04-22T14:16:36.485559+00:00  | INFO     |   
2024-04-22T14:16:36.50255+00:00   | INFO     |   
2024-04-22T14:16:36.518206+00:00  | INFO     |   CMake Warning at CMakeLists.txt:717 (message):
2024-04-22T14:16:36.533823+00:00  | INFO     |     Disabling OpenGL.
```
which is even more suspicious.
See https://github.com/jasper-software/jasper/blob/29154a52b6c2b895ac3db60d51be9f2a995ee145/CMakeLists.txt#L696-L720.

Reply:
`freeglut` has a header file `Library\include\GL\glut.h` (and `freeglut.h` too) in https://anaconda.org/anaconda/freeglut/3.0.0/download/win-64/freeglut-3.0.0-h6538335_5.tar.bz2 but not in https://anaconda.org/anaconda/freeglut/3.0.0/download/win-64/freeglut-3.0.0-hc56fc5f_5.tar.bz2. See https://anaconda.org/anaconda/freeglut/files

Reply:
Umm, using `freeglut 3.0.0 h6538335_5  # [win]` in `host` didn't change the log messages.

Reply:
conda-forge [found](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=906465&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=6119ccbe-9301-594f-7c27-f792b80a7fcc&l=1079) freeglut:
```
-- Found GLUT: D:/bld/jasper_1711830961660/_h_env/Library/lib/freeglut.lib
target: OpenGL::GL
target: OpenGL::GLU
target: GLUT::GLUT
GLUT::GLUT: adding include directory D:/bld/jasper_1711830961660/_h_env/Library/include
-- Looking for include file GL/glut.h
CMAKE_REQUIRE_INCLUDES D:/bld/jasper_1711830961660/_h_env/Library/include
-- Looking for include file GL/glut.h - found
```
I wonder if updating `freeglut` to `3.2.2` will help to resolve the issue

Reply:
Hm, `%PREFIX%\Library\include\GL` has `glut.h` in the `host` environment.
But     `-DGLUT_INCLUDE_DIR=%LIBRARY_INC%` didn't fix the issue

Reply:
You'll have to test on a windows machine and debug the cmake stuff to see where it's looking at and what's it's looking for.

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_DEBUG_MODE.html

Reply:
@JeanChristopheMorinPerso I tried `CMAKE_FIND_DEBUG_MODE` but it doesn't say any helpful. 
I changed folders, copied/renamed header `glut.h`/`freeglut.h` files, added `pkg-config` (it works only if **cmake <3.23**) but the result the same `glut.h not found`

Reply:
If I rename `copy "%LIBRARY_LIB%\pkgconfig\freeglut.pc" "%LIBRARY_LIB%\pkgconfig\glut.pc"`
then `pkg-config` says:
```
-- Found PkgConfig: C:/Users/dev/miniconda3/conda-bld/jasper_1714140635941/_build_env/Library/bin/pkg-config.exe (found version "0.29.2")
-- Checking for module 'glut'
--   Found glut, version 3.0.0
-- Found GLUT: TRUE
target: OpenGL::GL
target: OpenGL::GLU
target: GLUT::GLUT
GLUT::GLUT: adding include directory C:/Users/dev/miniconda3/conda-bld/jasper_1714140635941/_h_env/Library/include
CMAKE_REQUIRE_INCLUDES C:/Users/dev/miniconda3/conda-bld/jasper_1714140635941/_h_env/Library/include
-- Looking for include file GL/glut.h
-- Looking for include file GL/glut.h - not found
-- Looking for include file GLUT/glut.h
-- Looking for include file GLUT/glut.h - not found
-- Looking for include file glut.h
-- Looking for include file glut.h - not found
CMake Warning at CMakeLists.txt:716 (message):
  The glut.h header file is missing.
```

I wonder if there is a name collision issue.

Reply:
@skupr-anaconda do you have the logs for CMAKE_FIND_DEBUG_MODE? Usually it's very helpful and tells exactly what it tried. I can have a look at it.

Reply:
I've just pushed changes with `CMAKE_FIND_DEBUG_MODE`, cmake 3.22 pinning, and pkg-config.

Reply:
It would be slight better if you get logs for only the parts you want to debug. `CMAKE_FIND_DEBUG_MODE` is very very verbose. That's why the documentation at https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_DEBUG_MODE.html says to enable it only for the find calls you want to debug.

Reply:
Rafael pointed me out that we do not provide `jiv.exe `on Windows. I've check it here https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Fwin-64%2Fjasper-2.0.14-hea7d32e_2.conda. But we do provide `bin/jiv` on Unix platforms. So if I won't patch and fix it we can disable `freeglut` on `win`

Reply:
@skupr-anaconda I'm not sure to follow, jiv is from jasper isn't it? WHy would we not provide it on Windows when conda-forge provides it?

Reply:
@JeanChristopheMorinPerso Finally, fixed it on Windows by renaming files:
  - `pkgconfig\freeglut.pc` to **pkgconfig\glut.pc**
  - `freeglut.lib` to **glut.lib** and (_sic!_) **freeglutd.lib**

and setting CMake flags:
```
 -DGLUT_INCLUDE_DIR:PATH=%LIBRARY_INC% ^
 -DGLUT_glut_LIBRARY_RELEASE=%LIBRARY_LIB%\glut.lib ^
```

Reply:
That's our freeglut package which should be corrected. It provides freeglut.pc but not glut.pc. And freeglut.pc instructs to link to glut when the package only provides freeglut. The conda-forge recipe seems to have all that corrected.
Also, on the issue with GL/glut.h not being found. cmake check_include_files likely finds the file but fails to compile a simple test program. c-f has this patch on freeglut which I suspect would help with this issue: https://github.com/conda-forge/freeglut-feedstock/blob/main/recipe/disable_autolink_windows.patch

Reply:
That sounds fair. I'll look at freeglut. Thank you both @JeanChristopheMorinPerso and @cbouss!

Reply:
I've updated `freeglut` to 3`.4.0` with a patch that fixes the usage of `freeglut.lib` instead of freeglut**d**.lib